packages/wallet/src/test-utils/launchNode.test.ts

```diff
@@ -1,81 +1,192 @@
-import { Provider } from '@fuel-ts/providers';
-import path from 'path';
-import { cwd } from 'process';
+import { safeExec } from '@fuel-ts/errors/test-utils';
+import * as childProcessMod from 'child_process';
 
-import { WalletUnlocked } from '../wallets';
+import { killNode, launchNode } from './launchNode';
 
-import { launchNodeAndGetWallets } from './launchNode';
+type ChildProcessWithoutNullStreams = childProcessMod.ChildProcessWithoutNullStreams;
+
+vi.mock('child_process', async () => {
+  const mod = await vi.importActual('child_process');
+  return {
+    __esModule: true,
+    ...mod,
+  };
+});
+
+/**
+ * This should mimic the stderr.on('data') event, returning both
+ * success and error messages, as strings. These messages are like
+ * the ones from `fuel-core` startup log messages. We filter them
+ * to know fuel-core state.
+ */
+function mockSpawn(params: { shouldError: boolean } = { shouldError: false }) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const stderrOn = (eventName: string, fn: (data: any) => void) => {
+    if (eventName === 'data') {
+      if (params.shouldError) {
+        // The `IO error` message simulates a possible fuel-core error log message
+        fn('IO error');
+      } else {
+        // The `Binding GraphQL provider to` message simulates a fuel-core
+        // successful startup log message, usually meaning that the node
+        // is up and waiting for connections
+        fn('Binding GraphQL provider to');
+      }
+    }
+  };
+
+  const innerMocks = {
+    on: vi.fn(),
+    stderr: {
+      pipe: vi.fn(),
+      on: vi.fn(stderrOn),
+      removeAllListeners: vi.fn(),
+    },
+    stdout: {
+      pipe: vi.fn(),
+      removeAllListeners: vi.fn(),
+    },
+  };
+
+  const spawn = vi
+    .spyOn(childProcessMod, 'spawn')
+    .mockImplementation(
+      (..._: unknown[]) => innerMocks as unknown as ChildProcessWithoutNullStreams
+    );
+
+  return { spawn, innerMocks };
+}
+
+const defaultLaunchNodeConfig = {
```

Comment:
This could have a type annotation as well:

> _You'll have to export `LaunchNodeOptions`, which I also think is a good idea._

```ts
const defaultLaunchNodeConfig: Partial<LaunchNodeOptions> = {
```